cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -379,6 +457,109 @@ writeLedgerState mOutFile qState@(SerialisedLedgerState serLedgerState) =
    decodeLedgerState (SerialisedLedgerState (Serialised ls)) =
      first (const ls) (decodeFull ls)
 
+writeStakeSnapshot :: forall era ledgerera.
+                      ShelleyLedgerEra era ~ ledgerera
+                   => Era.Era ledgerera
+                   => FromCBOR (LedgerState era)
+                   => Hash StakePoolKey
+                   -> SerialisedLedgerState era
+                   -> ExceptT ShelleyQueryCmdError IO ()
+
+writeStakeSnapshot poolId qState =
+  case decodeLedgerState qState of
+    Left bs -> firstExceptT ShelleyQueryCmdHelpersError $ pPrintCBOR bs
+    Right ledgerState ->
+      if isNothing maybehk then
+        left $ ShelleyQueryCmdPoolIdError poolId
+      else
+        liftIO . LBS.putStrLn $  encodePretty $ Stakes {markpool = markStake, setpool = setStake, gopool = goStake, marktot = markTotal, settot = setTotal, gotot = goTotal}
+      where
```

Comment:
Let's move the functions/values from this where clause to the outer where clause. No need to have two.

Reply:
Again, this is not possible, so I have pushed the function definitions into the inner scope

Reply:
Scrap that.  decodeLedgerState is used to generate the Right so can't be pushed inwards.  I've lifted it to the module level instead